src/api.ts

```diff
@@ -232,3 +235,30 @@ export async function ensureDependencyInstalled(
 
   await addDependency(name, resolvedOptions);
 }
+
+export async function dedupeDependencies(options: Pick<OperationOptions, 'cwd' | 'silent'> & {recreateLockfile?: boolean} = {}) {
+  const resolvedOptions = await resolveOperationOptions(options);
+  const isSupported = ["bun", "deno"].includes(resolvedOptions.packageManager?.name ?? "");
+  const recreateLockfile = options.recreateLockfile ?? !isSupported;
+  if(!isSupported) {
+    consola.log(
+      `Deduping is currently not supported for \`${resolvedOptions.packageManager?.name ?? "unknown package manager"}\`.`,
```

Comment:
IMO it might only make users more confused, goal of nypm is transparent compatibility and recreate is already closest behavior + we log about what we do.

Also, it might make implicit issues, if a library passes `recreateLockfile: false` it will be a no-op function on deno/bun.

I suggest we do fallback without warn by default and in case `recreateLockfile: false` is passed, throw an error so we cover both max compat (default) and strict (if option explicitly asks to not fallback)

Reply:
so to summarise: 
|    | recreate | result |
| ---| ----------- | ----------- |
| deno or bun | false | error |
| deno or bun | true | recreate lockfile |
| any other pm | true | recreate lockfile |
| any other pm | false | dedupe |

Did I understand that correctly?

Reply:
Exactly. Without passing recreate (undefined)

|    | recreate | result |
| ---| ----------- | ----------- |
| deno or bun | undefined | recreate lockfile |
| any other pm | undefined | dedupe |
| deno or bun | false | error |
| any other pm | false | dedupe |
| deno or bun | true | recreate lockfile |
| any other pm | true | recreate lockfile |

Reply:
I tried to add tests as best as possbile. it was not possible for me to test for deno and bun with recreate false. Because the errors origin is the test.